### PR TITLE
Feature/nh prompt display

### LINKED ISSUE #31: Seed prompts
Collect the actual prompts, assign each one a genre, and add them to the seed file on the server.

### LINKED ISSUE #32: Displaying prompts on start of a new story

### PR DESCRIPTION
**Pull request by nickhartdev**
#### What's this PR do?
- Starts setting up ApiHelper
- Sets up writing prompt display to follow user stories (it can display one randomly selected from a given genre, all genres, or none at all)
- Adds some testing coverage for conditional renders
#### Where should the reviewer start?
ApiHelper.js, then StorySetup.js - check out handleSubmit
#### How should this be manually tested?
Click new story, select a prompt and see one show up! If the prompt option isn't selected, none should display.
#### What are the relevant tickets?
Resolves #31, resolves #32 
#### Screenshots (if appropriate)
Screen Shot 2020-09-13 at 11 25 50 PM
#### Questions:
- At this point I'm realizing the StorySetupView isn't doing too much, whereas StorySetup is doing a lot of the heavy lifting - we could refactor to have the former hold on to a few more methods/info, but everything's working as is. How could we leverage StorySetupView more? Lemme know what y'all think!
- Gonna make an issue on the BE repo, but there's a few prompt endpoints that come up with an empty response every once in a while.

### CODE REVIEW

File: src/components/StorySetup/StorySetup.js [hunk 1]
```diff
@@ -25,10 +26,16 @@ class StorySetup extends Component {
 
   handleSubmit = async (event) => {
     event.preventDefault()
-    // const prompt = await this.props.getPrompts(this.state.genre)
+    let prompt
+
+    if (this.state.promptRequested) {
```
Comment (GreysonElkins) [inline src/components/StorySetup/StorySetup.js:31]: I don't think this needs to be stored in state.

**Review by GreysonElkins — APPROVED** [review]
One teensy thing, but looks good!

File: src/components/StoryEditFields/StoryEditFields.test.js [hunk 2]
```diff
@@ -5,19 +5,29 @@ import StoryEditFields from './StoryEditFields'
...
+    render(
+      <Router>
+        <StoryEditFields prompt={{prompt: 'This is a prompt'}} />
+      </Router>
+    )
+    
+    const header = screen.getByText(/this is a prompt/i)
+
+    expect(header).toBeInTheDocument()
+  })
```
Comment (carlymclift) [inline src/components/StoryEditFields/StoryEditFields.test.js:36]: Nice testing here

**Review by carlymclift — APPROVED** [review]
Looks good! This will really start to pull things together

File: src/components/StorySetup/StorySetup.js (hunk 1 above)
Comment (nickhartdev) [reply src/components/StorySetup/StorySetup.js:31]: I needed some sort of conditional to keep the request from firing unless the box was checked - if you have another way I'd be happy to hear it

File: src/ApiHelper/ApiHelper.js [hunk 3]
```diff
@@ -1,6 +1,19 @@
...
+
+    if (genre === 'any') {
+      response = await fetch(baseUrl + '/prompts/any')
+    } else {
+      response = await fetch(baseUrl + `/prompts/${genre}`)
+    }
+    const prompt = await response.json()
+    return prompt
+  }
+
```
Comment (Abdeboskey) [inline src/ApiHelper/ApiHelper.js:16]: do we want to add some of the back end error handling here?

File: src/components/StoryEditFields/StoryEditFields.js [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
 
-const StoryEditFields = ({ updateText, disabled }) => {
+const StoryEditFields = ({ updateText, disabled, prompt }) => {
   return (
     <Fragment>
-      <h1 id="prompt">Prompt placeholder</h1>
+      {prompt && 
+        <h1 id="prompt">{ prompt.prompt }</h1>
+      }
       <input
```
Comment (Abdeboskey) [inline src/components/StoryEditFields/StoryEditFields.js:10]: I dig this!

File: src/components/StoryEditView/StoryEditView.js [hunk 5]
```diff
@@ -21,6 +21,7 @@ class StoryEditView extends Component {
     return (
       <main id="story-edit-container">
         <StoryEditFields 
+          prompt={ this.props.location.state.prompt }
           updateText={ this.updateText }
```
Comment (Abdeboskey) [inline src/components/StoryEditView/StoryEditView.js:25]: So you're telling me this works!?

File: src/components/StorySetup/StorySetup.js (hunk 1 above)
Comment (Abdeboskey) [reply src/components/StorySetup/StorySetup.js:31]: That was my thought as well, just to use it as a flag to trigger the request. but yes, if there's another way I'd be happy to know!

File: src/components/StorySetup/StorySetup.js [hunk 6]
```diff
@@ -78,9 +85,8 @@ class StorySetup extends Component {
             onChange={this.handleChange}
           >
             <option value='any'>Any</option>
-            <option value='dystoptian'>Dystopian</option>
+            <option value='dystopian'>Dystopian</option>
             <option value='fantasy'>Fantasy</option>
```
Comment (Abdeboskey) [inline src/components/StorySetup/StorySetup.js:89]: good catch

**Review by Abdeboskey — APPROVED** [review]
Looking good, thanks Nick!

**Pull request merged**